pkg/tracing/logger.go

```diff
@@ -50,21 +53,37 @@ func NewLogger(logger *zap.Logger, tracer trace.Tracer, attrs ...attribute.KeyVa
 }
 
 func (l traceLogger) StartSpan(ctx context.Context, op string, attrs ...attribute.KeyValue) (context.Context, ScopedLogger) {
+	childCtx, childSpan := l.Tracer.Start(ctx, op)
+
+	l.addAttributesToSpan(childCtx, childSpan, attrs...)
+
+	return NewScopedLogger(childCtx, childSpan, l)
+}
+
+func (l traceLogger) StartSpanNamedByCaller(ctx context.Context, attrs ...attribute.KeyValue) (context.Context, ScopedLogger) {
+	childCtx, childSpan := l.Tracer.Start(ctx, callerFunc(0))
+
+	l.addAttributesToSpan(childCtx, childSpan, attrs...)
+
+	return NewScopedLogger(childCtx, childSpan, l)
+}
+
+func (l traceLogger) addAttributesToSpan(ctx context.Context, span trace.Span, attrs ...attribute.KeyValue) {
 	type HasParentSpan interface {
 		ParentSpanID() trace.SpanID
 	}
-	childCtx, childSpan := l.Tracer.Start(ctx, op)
-	childSpan.SetAttributes(attrs...)
+
+	span.SetAttributes(attrs...)
 
 	isRootSpan := true
-	if s, ok := childSpan.(HasParentSpan); ok {
+	if s, ok := span.(HasParentSpan); ok {
 		isRootSpan = !s.ParentSpanID().IsValid()
 	}
+	// TODO: why do we set root span attributes onto child span as well?
```

Comment:
That's not what this is doing. When we create a TraceLogger, there are some attributes that we want set on the root span (the very first span that is created by the cli). This checks if we just made the first span (root span) and sets attributes on the root. If this is a child span, nothing is set on it.

In particular, I am generating a correllation id that should be set on the root span, so that when we create a span in another process (like inside the invocation image) it can use the same correllation id so we can trace across system.